lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -90,18 +89,18 @@ def latent_heat_of_condensation(temperature_input):
         vapour using the relationship employed by the UM.
 
         Args:
-            temperature_input (iris.cube.Cube):
+            temperature_input (numpy.ndarray):
                 A cube of air temperatures (Celsius, converted if not).
         Returns:
-            iris.cube.Cube:
+            numpy.ndarray:
                 Temperature adjusted latent heat of condensation (J kg-1).
         """
-        temperature = temperature_input.copy()
-        temperature.convert_units('celsius')
-        latent_heat = (-1. * consts.U_LATENT_HEAT_T_DEPENDENCE * temperature +
-                       consts.U_LH_CONDENSATION_WATER)
-        latent_heat.units = consts.U_LH_CONDENSATION_WATER.units
-        latent_heat.rename('latent_heat_of_condensation')
+        # TODO Do this without using Unit. problem is float64 error.
+        unit = Unit('kelvin')
+        temperature = unit.convert(temperature_input, 'celsius')
```

Comment:
The temperature.convert_units('celsius')  will only converts to Celsius if the cube is not already in celsius. This change will convert regardless of the units of the data. We have two choices. We can pass the units of the data in the input array as an extra variable or insist that the input data is in a particular unit. All upstream calls to this function would need to ensure the data is in the correct units.

Reply:
If Unit is causing problems and we can guarantee incoming data is in Kelvin you could just make use of ABSOLUTE_ZERO as you do further down in the code.

Reply:
Most of the code I have seen everything is in kelvin, so should that be standardised?
The conversion problem occused here even when I wrote it the same as further down the code. A float64 error that I havn't been able to work out.